module_prototyper/wizard/module_prototyper_module_export.py

```diff
@@ -0,0 +1,132 @@
+# -*- encoding: utf-8 -*-
+# #############################################################################
+#
+# OpenERP, Open Source Management Solution
+# This module copyright (C) 2010 - 2014 Savoir-faire Linux
+# (<http://www.savoirfairelinux.com>).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import StringIO
+import base64
+import os
+import zipfile
+from collections import namedtuple
+
+from openerp import fields, models, api
+
+
+class PrototypeModuleExport(models.TransientModel):
+    _name = "module_prototyper.module.export"
+
+    name = fields.Char('File Name', readonly=True)
+    api_version = fields.Selection(
+        [
+            ('8.0', '8.0'),
+        ],
+        'API version',
+        required=True,
+        default='8.0'
+    )
+    data = fields.Binary('File', readonly=True)
+    state = fields.Selection(
+        [
+            ('choose', 'choose'),  # choose version
+            ('get', 'get')  # get module
+        ],
+        default='choose'
+    )
+
+    @api.model
+    def action_export(self, ids):
+        """
+        Export a zip file containing the module based on the information
+        provided in the prototype, using the templates chosen in the wizard.
+        """
+        if isinstance(ids, (int, long)):
+            ids = [ids]
+        wizard = self.browse(ids)
+
+        active_model = self._context.get('active_model')
+
+        # checking if the wizard was called by a prototype.
+        msg = '{} has to be called from a "module_prototyper" , not a "{}"'
+        assert active_model == 'module_prototyper', msg.format(
+            self, active_model
+        )
+
+        # getting the prototype of the wizard
+        prototypes = self.env[active_model].browse(
+            [self._context.get('active_id')]
```

Comment:
Why not naming this variable prototype? browsing from active_id should return only a record set of size 1.

Plus, you can pass a single id instead of a list.